core/src/main/java/com/novoda/noplayer/exoplayer/ExoPlayerFacade.java

```diff
@@ -140,26 +140,30 @@ private void setExoPlayerTextOutput(TextRenderer.Output output) {
     }
 
     public void selectSubtitleTrack(PlayerSubtitleTrack subtitleTrack) {
-        subtitleTrackSelector.selectTextTrack(subtitleTrack);
+        subtitleTrackSelector.selectTextTrack(subtitleTrack, rendererTypeRequester);
     }
 
     public List<PlayerSubtitleTrack> getSubtitleTracks() {
-        return subtitleTrackSelector.getSubtitleTracks();
+        return subtitleTrackSelector.getSubtitleTracks(rendererTypeRequester);
     }
 
     public boolean hasPlayedContent() {
         return exoPlayer != null;
     }
 
-    public SimpleExoPlayer getRawExoPlayer() {
-        return exoPlayer;
-    }
-
     public void clearSubtitleTrack() {
-        subtitleTrackSelector.clearSubtitleTrack();
+        subtitleTrackSelector.clearSubtitleTrack(rendererTypeRequester);
     }
 
     public void selectFirstAvailableSubtitlesTrack() {
-        subtitleTrackSelector.selectFirstTextTrack();
+        subtitleTrackSelector.selectFirstTextTrack(rendererTypeRequester);
     }
+
+    @VisibleForTesting
+    final RendererTypeRequester rendererTypeRequester = new RendererTypeRequester() {
```

Comment:
We need to call the native `exoPlayer.getRendererType` that is instantiated after its creation, we do not have it in the factory. Happy to pair with you on this to find a better solution.